k8s/helm/cortex/values.yaml

```diff
@@ -0,0 +1,165 @@
+## Affinity for pod assignment
+## ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+affinity: {}
+# podAntiAffinity:
+#   requiredDuringSchedulingIgnoredDuringExecution:
+#   - labelSelector:
+#       matchExpressions:
+#       - key: app
+#         operator: In
+#         values:
+#         - cortex
+#     topologyKey: "kubernetes.io/hostname"
+
+## StatefulSet annotations
+annotations: {}
+
+# enable tracing for debug, need install jaeger and specify right jaeger_agent_host
+tracing:
+  jaegerAgentHost:
+
+config:
+  auth_enabled: false
+  distributor:
+    shard_by_all_labels: true
+    pool:
+      health_check_ingesters: true
+  ingester_client:
+    grpc_client_config:
+      # Configure the client to allow messages up to 100MB.
+      max_recv_msg_size: 104857600
+      max_send_msg_size: 104857600
+      use_gzip_compression: true
+  ingester:
+    #chunk_idle_period: 15m
+    lifecycler:
+      address: 127.0.0.1
+      # We want to start immediately and flush on shutdown.
+      join_after: 0
+      claim_on_rollout: false
+      final_sleep: 0s
+      num_tokens: 512
+      # Use an in memory ring store, so we don't need to launch a Consul.
+      ring:
+        kvstore:
+          store: inmemory
+        replication_factor: 1
+  # Use local storage - BoltDB for the index, and the filesystem
+  # for the chunks.
+  schema:
+    configs:
+    - from: 2019-03-25
+      store: boltdb
+      object_store: filesystem
+      schema: v10
+      index:
+        prefix: index_
+        period: 168h
+  server:
+    http_listen_port: 9009
+    grpc_server_max_recv_msg_size: 104857600
+    grpc_server_max_send_msg_size: 104857600
+    grpc_server_max_concurrent_streams: 1000
+  storage:
+    boltdb:
+      directory: /data/cortex/index
+    filesystem:
+      directory: /data/cortex/chunks
+
+image:
+  repository: quay.io/cortexproject/cortex
+  tag: master-d0f8944c
+  pullPolicy: Always # Always pull while in BETA
+
+## Additional Cortex container arguments, e.g. log level (debug, info, warn, error)
+extraArgs: {}
+  # log.level: debug
+
+livenessProbe:
+  httpGet:
+    path: /ready
+    port: http-metrics
+  initialDelaySeconds: 45
+
+
+## ref: https://kubernetes.io/docs/user-guide/node-selection/
+nodeSelector: {}
+
+## ref: http://kubernetes.io/docs/user-guide/persistent-volumes/
+## If you set enabled as "True", you need :
+## - create a pv which above 10Gi and has same namespace with cortex
```

Comment:
Since the statefulset is using a volume claim template when there is no existing claim specified, I think line 90 is not accurate. The only requirement I understand in this case is to have a storage class defined, as the pv & claim will be generated automatically from the template.

Reply:
you're right the comment is outdated. I'll remove it. For the test I'm not sure but I think this was a flaky test.

Reply:
Yes, you are right about the test. It was corrected in a different commit